rpc/src/transport/channel.rs

```diff
@@ -113,33 +112,25 @@ mod tests {
             Ok::<_, io::Error>((response1, response2))
         };
 
-        let (response1, response2) = run_future(future::join(
+        let (response1, response2) = runtime::task::spawn(future::join(
```

Comment:
The `join` here was only to make it easier to run the future. Since we're in an async fun, you can just spawn the server first and then await each response individually, like we do in other tests.